lib/chef/provider/registry_key.rb

```diff
@@ -91,6 +98,18 @@ def define_resource_requirements
           a.failure_message(Chef::Exceptions::Win32RegNoRecursive, "#{@new_resource.key} has subkeys but recursive is set to false.")
           a.whyrun("#{@current_resource.key} has subkeys, but recursive is set to false. attempt to delete would fails unless subkeys were deleted prior to this action.")
         end
+        requirements.assert(:create, :create_if_missing) do |a|
+          #If type key missing in the RegistryKey values hash
+          a.assertion { !key_missing?(@new_resource.values, :type) }
+          a.failure_message(Chef::Exceptions::RegKeyValuesTypeMissing, "Missing type key in RegistryKey values hash")
+          a.whyrun("Type key does not exist. Attempt would fail unless the complete values hash containing all the keys does not exist for registry_key resource's create action.")
+        end
+        requirements.assert(:create, :create_if_missing) do |a|
+          #If data key missing in the RegistryKey values hash
```

Comment:
A space after the # would improve readability, e.g. `# If the data key is missing in the RegistryKey values hash`